contracts/dapps/astar-dapp-staking/AstarDappStakingFactory.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.0;
+pragma experimental ABIEncoderV2;
+
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../../interfaces/IPoolFactory.sol";
+import "./AstarDappStaking.sol";
+
+/**
+ * @dev Factory contract of Astar Dapp staking pool.
+ *x
+ */
+contract AstarDappStakingFactory is IPoolFactory, Ownable {
+    using BytesLib for bytes;
+
+    address astarDAppStakingContract;
+
+    event StakingContractChange(address indexed oldContract, address indexed newContract);
+    event AStarDappStakingCreated(
+        address indexed pool,
+        address indexed community,
+        string name,
+        address dapp
+    );
+
+    constructor(address _astarDAppStakingContract) {
+        require(_astarDAppStakingContract != address(0), 'Invalid argument');
+        astarDAppStakingContract = _astarDAppStakingContract;
+    }
+
+    function adminSetStakingContract(address _astarDAppStakingContract) external onlyOwner {
```

Comment:
Is this will be changed? If this address will change, I think the old pools created by this factory will change too. maybe the sub pools can read this contract address from this factory.

Reply:
No, old pool still can use the old one. If AStar change it, we need to use the new one.